### ISSUE TITLE
Remove validation in QueryBuilder (or add a way to disable it)

### ISSUE THREAD

**Issue by olim7t** [stargate/stargate#664]
`QueryBuilder` performs preliminary checks against the schema as the query is being built. I've run into various cases where this is really annoying and counter-productive.

For example `create()` queries fail with `IllegalArgumentException` if the element already exists.

* Let's say I'm handling a user request to create a keyspace with specific replication options, but if it already exists with different options I want to fail and show the differences. I can't use `ifNotExist()` (it doesn't take options into account), so I need to execute a regular `create()` and catch the error. But `IllegalArgumentException` is very generic, I'll have to rely on some brittle message parsing. I would much rather deal with the persistence backend's `AlreadyExistsException`. And in fact I'll have to deal with that exception as well: there could easily be a race condition where the keyspace gets created after the query builder's check (especially if I don't use the built query immediately). So now I have to catch two exceptions.
* Another example is if I'm doing a CQL migration. I know the target data model, and if a table doesn't match my strategy is to drop and recreate it. It might look like I can build the queries as I'm iterating the schema, and execute them all at the end. But in fact that doesn't work: the `create()` query will fail because the table already exists.

Another problem is that some of those checks are wrong. For example in `BuiltUpdate`:

```java
    Preconditions.checkArgument(
        !rowsUpdated.isRanges(), "Invalid WHERE clause for UPDATE: cannot UPDATE a range of rows");
```

Range updates are valid in CQL.

`QueryBuilder` should focus on its primary responsibility: building queries. There might be a few runtime checks for syntactic errors that can't be enforced by the DSL. But semantic checks are the persistence backend's job.

IMHO those checks should be removed. If this is too aggressive, we could also add an option to disable them: `dataStore.queryBuilder(boolean strict)`.

**Comment by olim7t** [comment]
Related: `Column.Type.fromCqlDefinitionOf()` checks that UDTs exist in the keyspace. There are legitimate case where we might want to parse types that will only exist in the future, for example a tool that generates a schema according to a model.

**Comment by olim7t** [comment]
I addressed `fromCqlDefinitionOf ` in ##679.

**Comment by olim7t** [comment]
I took another look at this and I realized I had missed something: connecting the query builder to the schema was introduced by 2c4dbed96aa7410c46ebd80963d70cdeccb9ca32, and it's not just for semantic checks: it's also required to allow built queries to be inspected.

For example, given an INSERT statement, the query on its own is not enough to know which values are part of the primary key, we need to have the table's schema.

So even though we could still remove some duplicate checks, it's not possible to make the query builder completely disconnected from the schema. Querying a table will always require that the table exists at the time of query construction.
